src/oops/SOLID/singleResponsibilityPrinciple/before/EmployeeRepository.java

```diff
@@ -16,4 +20,24 @@
 
         return Arrays.asList(anna, billy, steve, magda);
     }
+
+    /**
+     * Persists the employee
+     * @param employee object to be persisted
+     */
+    public void save(Employee employee) throws Exception {
+        try {
+            String serializedEmployee = EmployeeFileSerializer.serialize(employee);
+
+            Path path = Paths.get(employee.getFullName()
+                    .replace(" ","_") + ".rec");
+            Files.write(path, serializedEmployee.getBytes());
+
+            System.out.println("Saved employee " + employee.toString());
+        } catch (IOException e){
```

Comment:
The main responsibility of the function is to save employee details, and the exception thrown while saving should be handled at the caller level.

Reply:
I have caught IOException specifically here, because this guy deals with the IO part. This will log the exception message, and further throw the exception to its caller as well, where also it is being handled and logged properly.